pkg/chain/message.go

```diff
@@ -53,7 +53,6 @@ type MessageFactory interface {
 // Actual message construction is delegated to a `MessageFactory`, and the message are opaque to the producer.
 type MessageProducer struct {
 	factory       MessageFactory
-	messages      []interface{}
```

Comment:
We should keep this so the producer is still capable of exporting all the messages at the end (for comparison with other impls, replay, ...)